marimbabot_planning/src/lilypond_to_hitpoint.cpp

```diff
@@ -106,87 +107,110 @@ std::vector<std::tuple<geometry_msgs::PoseStamped, double, double>> lilypond_to_
 }
 
 
-int main(int argc, char **argv)
+
+class SubscribeAndPublish
 {
-    ros::init(argc, argv, "marimba_move_lilypond_input_viz");
-    ros::NodeHandle node_handle;
-    ros::AsyncSpinner spinner(1);
-    spinner.start();
+public: 
+    SubscribeAndPublish() {
+        //Topic you want to publish
+        hit_marker_pub = n_.advertise<visualization_msgs::Marker>("hit_marker", 1);
 
-    // Create publisher for the hit marker
-    ros::Publisher hit_marker_pub = node_handle.advertise<visualization_msgs::Marker>("hit_marker", 1);
+        //Topic you want to subscribe
+        sub_ = n_.subscribe("vision_node/recognized_sentence", 1, &SubscribeAndPublish::moveToPoseCallback, this);
+    }
 
-    // Create tf2 listener
-    std::shared_ptr<tf2_ros::Buffer> tfBuffer = std::make_shared<tf2_ros::Buffer>();
-    tf2_ros::TransformListener tfListener(*tfBuffer);
+    void moveToPoseCallback(const std_msgs::String::ConstPtr& lilypond_sentence) {
+        ROS_DEBUG("moveToPoseCallback heard: [%s]", lilypond_sentence->data.c_str());
 
-    // Sleep to make sure the tf listener is ready and all publishers are connected
-    ros::Duration(1.0).sleep();
 
-    ROS_INFO("Starting marimba_move");
+        // Create tf2 listener
+        std::shared_ptr<tf2_ros::Buffer> tfBuffer = std::make_shared<tf2_ros::Buffer>();
+        tf2_ros::TransformListener tfListener(*tfBuffer);
 
-    std::string dummy_lilypond = "c1 r1 d4 e4 f4 g4 c2 e'2 e''1";
+        // Sleep to make sure the tf listener is ready and all publishers are connected
+        ros::Duration(1.0).sleep();
 
-    // Convert lilypond sequence to cartesian poses and times
-    auto hits = lilypond_to_cartesian(
-        tfBuffer,
-        "base_link",
-        dummy_lilypond
-        );
+        ROS_INFO("Starting marimba_move");
 
-    ROS_INFO("Converted lilypond sequence to cartesian poses and times");
+        std::string lilypond_string = lilypond_sentence->data.c_str();
 
-    // Loop over all hits and publish a marker at the correct time
-    double time = 0.0;
+        // Convert lilypond sequence to cartesian poses and times
+        auto hits = lilypond_to_cartesian(
```

Comment:
Optimally the result of this function would be directly published and the viz is a different node.

Reply:
https://github.com/UHHRobotics22-23/marimbabot/issues/63